x86/proofs/x86.ml

```diff
@@ -2746,10 +2777,11 @@ let X86_CONV (decode_ths:thm option array) ths tm =
      GEN_REWRITE_CONV (RAND_CONV o TOP_DEPTH_CONV) [GSYM ADD_ASSOC] THENC
      RAND_CONV NUM_REDUCE_CONV) THENC
    TOP_DEPTH_CONV COMPONENT_WRITE_OVER_WRITE_CONV THENC
-   GEN_REWRITE_CONV (SUB_COMPONENTS_CONV o TOP_DEPTH_CONV) ths THENC
+   GEN_REWRITE_CONV TOP_DEPTH_CONV ths THENC
```

Comment:
Question: I made this rewrite rule more general because there is an if-condition for MOVAPS/MOVDQA which checks alignment. Rewrites using the assumptions is required for simplifying that if-condition. However, the original version is more constrained in that it only concerns internals of `write` and `read` expressions. I'm wondering if this change is okay.

Reply:
Okay, this change is indeed a bad one as shown in the CI run that failed the x86 proofs build. With a second thought, I think it is the best to put `ALIGNED_16_CONV` in `X86_FORCE_CONDITIONAL_CONV`, since this conversion is used for reducing if-conditions. See [commit](https://github.com/awslabs/s2n-bignum/pull/212/commits/8324ae0cf95481af70298c38076dd63fed136fb9).